src/worker/ingestion/organization-manager.ts

```diff
@@ -0,0 +1,47 @@
+import { RawOrganization } from '../../types'
+import { DB } from '../../utils/db/db'
+import { timeoutGuard } from '../../utils/db/utils'
+
+type OrganizationCache<T> = Map<RawOrganization['id'], [T, number]>
+
+export class OrganizationManager {
+    db: DB
+    organizationCache: OrganizationCache<RawOrganization | null>
+
+    constructor(db: DB) {
+        this.db = db
+        this.organizationCache = new Map()
+    }
+
+    public async fetchOrganization(organizationId: RawOrganization['id']): Promise<RawOrganization | null> {
+        const cachedOrganization = this.getByAge(this.organizationCache, organizationId)
+        if (cachedOrganization) {
+            return cachedOrganization
+        }
+
+        const timeout = timeoutGuard(`Still running "fetchOrganization". Timeout warning after 30 sec!`)
+        try {
+            const organizationQueryResult = await this.db.postgresQuery(
```

Comment:
genuine q: we have SQL queries spread in multiple areas of the codebase, `db`, `sql`, and individual queries in some files here and there. do we have any sort of convention for where they should be?

Reply:
I think for some more bespoke stuff (like `TeamManager.updateEventNamesAndProperties`), it doesn't make sense to make it common via `DB`, BUT thanks for bringing this line up – I updated it to use `DB.fetchOrganization` (and similarly in `TeamManager` with `fetchTeam`)